pkg/subctl/datafile/datafile.go

```diff
@@ -73,7 +74,7 @@ func NewFromFile(filename string) (*SubctlData, error) {
 	return NewFromString(string(dat))
 }
 
-func NewFromCluster(restConfig *rest.Config, brokerNamespace string) (*SubctlData, error) {
+func NewFromCluster(restConfig *rest.Config, brokerNamespace string, enableLighthouse bool) (*SubctlData, error) {
```

Comment:
This is supposed to retrieve the information stored in the cluster, which doesn’t cover service discovery; I think we should avoid adding a parameter here, and set the value explicitly after the call to `NewFromCluster`:

    subctlData, err := datafile.NewFromCluster(config, broker.SubmarinerBrokerNamespace)
    subctlData.ServiceDiscovery = enableServiceDiscovery

Reply:
Either that, or make the function find out from the broker that it needs to enable service discovery.

Reply:
Going with setting value after call. Need to add it to subctlData so it shows up in `broker-info.subm`